packages/fast-tooling-react/src/css-editor/border-radius/border-radius.tsx

```diff
@@ -0,0 +1,343 @@
+import React from "react";
+import { get, pick } from "lodash-es";
+import Foundation, {
+    FoundationProps,
+    HandledProps,
+} from "@microsoft/fast-components-foundation-react";
+import {
+    BorderRadiusValue,
+    CSSBorderRadiusHandledProps,
+    CSSBorderRadiusProps,
+    CSSBorderRadiusState,
+    CSSBorderRadiusUnhandledProps,
+    CSSBorderRadiusValues,
+} from "./border-radius.props";
+
+export default class CSSBorderRadius extends Foundation<
+    CSSBorderRadiusHandledProps,
+    CSSBorderRadiusUnhandledProps,
+    CSSBorderRadiusState
+> {
+    public static displayName: string = "CSSBorderRadius";
+
+    protected handledProps: HandledProps<CSSBorderRadiusHandledProps> = {
+        data: void 0,
+        onChange: void 0,
+        managedClasses: void 0,
+    };
+
+    private parsedString: string[];
+
+    constructor(props: CSSBorderRadiusProps) {
+        super(props);
+
+        this.state = {
+            indvidualValues: false,
+            hasFocus: void 0,
+            topLeftValue: "0",
+            topRightValue: "0",
+            bottomRightValue: "0",
+            bottomLeftValue: "0",
```

Comment:
I would rather we do not store any of the values in state, i wouldn't store data, or individual values. A better pattern would be to have a memoized getter function that when passed in with the border radius could return these values to you. This way you never have to deal with state management and using `componentDidUpdate`